src/Scope/Free.hs

```diff
@@ -121,15 +121,15 @@ searchType :: ([Type] -> Type -> Bool) -> Ctxt -> Type -> Bool
 searchType pred g = h [] where
   h :: [Type] -> Type -> Bool
   h visited tp = pred visited tp || case tp of
-    TpData y as ->
+    TpData y tgs tis ->
       -- Don't search the same type twice (that would cause infinite recursion)
       not (tp `elem` visited) &&
       case ctxtLookupType2 g y of
         Nothing -> False
-        Just (ps, cs) ->
-          -- Substitute actual type parameters (as) for datatype's type parameters (ps)
+        Just (tgvars, tpvars, cs) ->
```

Comment:
Probably I should have written `TpData y tgs as` at line 124, but what should go here? `(tgs', ps, cs)`?